pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -252,15 +259,45 @@ func (c *Cluster) startProvisioningOverPVCs(config *provisionConfig) {
 		return
 	}
 
-	for _, volume := range c.ValidStorage.VolumeSources {
+	for i, volume := range c.ValidStorage.VolumeSources {
+		metadataDevicePVCSource := v1.PersistentVolumeClaimVolumeSource{}
+
+		// If the volumeTemplate is unknown we do nothing
+		if volume.Type != bluestorePVCMetadata && volume.Type != bluestorePVCBlock {
+			logger.Errorf("unknown PVC template type %q, valid names are %q for the main OSD block and %q for a metadata device to back the OSD.", volume.Type, bluestorePVCBlock, bluestorePVCMetadata)
+			continue
+		}
+
+		// We don't need to use the metadata devices as OSDs
+		// They are just attached to OSD and field in their property
+		if volume.Type == bluestorePVCMetadata {
+			logger.Infof("PVC %q is not an OSD but a %q device", volume.PersistentVolumeClaimSource.ClaimName, volume.Type)
+			continue
+		}
+
+		// Let's see how the next PVC looks like
+		// If the next PVC has been identified as a PVC let's attached to this OSD property
+		//
+		// The logic will get a bit more complex if we plan support for a third device for "block.wal"
```

Comment:
Can we not rely on the `VolumeSources` slide ordering to associate the data and metadata PVCs? It feels like we need another struct to help keep them associated correctly. To keep it simple, maybe we add a field to VolumeSources, which is the metadataDevice VolumeSource.

Reply:
Won't this be a recursive inclusion then? `VolumeSources` is just a list of `VolumeSource` so I'm not sure this works unless I missed something.

Reply:
Let's stick with what you have. As long as the slice order is used consistently we should be fine.